app/grandchallenge/reader_studies/forms.py

```diff
@@ -310,6 +337,52 @@ def interface_choices(self):
     def initial_interface(self):
         return self.interface_choices().first()
 
+    def widget_choices(self):
+        answer_type = self["answer_type"].value()
+        choices = [("", "Default")]
+        if answer_type:
+            try:
+                extra_options = ANSWER_TYPE_TO_QUESTION_WIDGET[answer_type]
+                choices.extend(
+                    (option.name, option.label) for option in extra_options
+                )
+            except KeyError:
+                pass
+        return choices
+
+    def initial_widget(self):
+        try:
+            return self.instance.widget.kind
+        except ObjectDoesNotExist:
+            return ""
+
+    def clean_widget(self):
+        value = self.cleaned_data.get("widget")
+        if value == "":
+            return None
+        else:
+            widget_instance = QuestionWidget(
+                kind=value, question=self.instance
+            )
+
+            if value == QuestionWidgetKindChoices.ACCEPT_REJECT:
+                if not self.cleaned_data.get("interface"):
+                    self.add_error(
+                        error=ValidationError(
+                            f"To use the {widget_instance.get_kind_display()} widget you must define a default answer."
+                        ),
+                        field=None,
+                    )
+                if self.cleaned_data.get("required"):
+                    self.add_error(
+                        error=ValidationError(
+                            f'To use the {widget_instance.get_kind_display()} widget you must uncheck "required".'
+                        ),
+                        field=None,
+                    )
```

Comment:
OK, now things are getting clearer 🙂

Because this validation depends on two fields (widget and interface, and widget and required) in the form this should not be done in `clean_widget`, it must be done in `clean`. See [the Django docs](https://docs.djangoproject.com/en/4.1/ref/forms/validation/#raising-validation-error). 

> The form subclass’s clean() method can perform validation that requires access to multiple form fields. This is where you might put in checks such as “if field A is supplied, field B must contain a valid email address”.